### PR TITLE
react-combobox: add separate Dropdown component

### PR DESCRIPTION
**Pull request by smhigley**
This adds a `Dropdown` component to the `react-combobox` package, and updates `Combobox` to be an editable field with an `<input>` as the primary slot.

This PR depends on the changes in #23499

Note: there are some quirks with how the `input` behaves when selecting vs. typing that will be fixed in a separate PR.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                |     Baseline (minified/GZIP) |                           PR |                                                                                                                                                                                                  Change |
| :----------------------------------------------------------------------------------------------- | ---------------------------: | ---------------------------: | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox</abbr> | `61.189 kB`<br />`20.959 kB` | `61.468 kB`<br />`21.054 kB` | `279 B` <br />`95 B`  |

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/82337ca348673485e78b9072431c00c222cd985a'>82337ca348673485e78b9072431c00c222cd985a</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-combobox/src/components/Dropdown/useDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      type: 'button',
+      children: baseState.value || props.placeholder,
+      ...triggerNativeProps,
+    },
+  });
+
+  const listboxShorthand = resolveShorthand(props.listbox, { required: true });
+
+  const [triggerWithPopup, listboxWithPopup] = useComboboxPopup(props, triggerShorthand, listboxShorthand);
+  const [triggerSlot, listboxSlot] = useTriggerListboxSlots(props, baseState, ref, triggerWithPopup, listboxWithPopup);
```
Comment (layershifter) [inline packages/react-components/react-combobox/src/components/Dropdown/useDropdown.tsx:40]: As I mentioned in the other PR: probably these hooks should accept and return `state` to be consistent with other components.

**Comment by layershifter** [comment]
> Bundle size report

Would you mind to add a bundle size fixture for `Dropdown`? 🐱

File: packages/react-components/react-combobox/src/components/Combobox/useComboboxStyles.ts [hunk 2]
```diff
@@ -16,10 +16,14 @@ export const comboboxClassNames: SlotClassNames<ComboboxSlots> = {
  */
 const useStyles = makeStyles({
   root: {
+    alignItems: 'center',
     ...shorthands.border(tokens.strokeWidthThin, 'solid', 'transparent'),
     ...shorthands.borderRadius(tokens.borderRadiusMedium),
     boxSizing: 'border-box',
-    display: 'inline-block',
+    columnGap: tokens.spacingHorizontalXXS,
```
Comment (micahgodbolt) [inline packages/react-components/react-combobox/src/components/Combobox/useComboboxStyles.ts:23]: gap

File: packages/react-components/react-combobox/src/components/Dropdown/useDropdown.tsx (hunk 1 above)
Comment (smhigley) [reply packages/react-components/react-combobox/src/components/Dropdown/useDropdown.tsx:40]: I actually tried doing this initially, but the problem is essentially with typing. The issue is that Combobox vs. Dropdown have different and incompatible primary slot types (button vs. input, with different children types), and I didn't find a way to type the state prop and return type in a way that would work without a bunch of explicit `as X` casting.

I'd be happy to work with you on that more, if you have any thoughts on how to make that work. For now, both hooks are fully internal and not exported, so we can change this as much as we want without affecting the public API :)

File: packages/react-components/react-combobox/src/components/Dropdown/Dropdown.types.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  ComboboxBaseContextValues,
+  ComboboxBaseOpenChangeData,
+  ComboboxBaseOpenEvents,
+  ComboboxBaseProps,
+  ComboboxBaseState,
+} from '../../utils/ComboboxBase.types';
+import { Listbox } from '../Listbox/Listbox';
+
+export type DropdownSlots = {
+  /* The root combobox slot */
```
Comment (GeoffCoxMSFT) [inline packages/react-components/react-combobox/src/components/Dropdown/Dropdown.types.ts:12]:
```suggestion
  /* The root dropdown slot */
```

File: packages/react-components/react-combobox/src/stories/DropdownDefault.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import * as React from 'react';
+import { useId } from '@fluentui/react-utilities';
+import { makeStyles, shorthands } from '@griffel/react';
+import { Dropdown, DropdownProps, Option } from '../index';
```
Comment (TristanWatanabe) [inline packages/react-components/react-combobox/src/stories/DropdownDefault.stories.tsx:4]: Same here
```suggestion
import { Dropdown, DropdownProps, Option } from '@fluentui/react-combobox';
```

**Review by GeoffCoxMSFT — APPROVED** [review]

**Review by micahgodbolt — APPROVED** [review]

File: packages/react-components/react-combobox/src/utils/ComboboxBase.types.ts [hunk 5]
```diff
@@ -33,7 +34,7 @@ export type ComboboxBaseProps = SelectionProps & {
   /**
    * Callback when the open/closed state of the dropdown changes
    */
-  onOpenChange?: (e: ComboboxOpenEvents, data: ComboboxOpenChangeData) => void;
+  onOpenChange?: (e: ComboboxBaseOpenEvents, data: ComboboxBaseOpenChangeData) => void;
```
[the remark this answers is not in the document]
Comment (smhigley) [reply packages/react-components/react-combobox/src/utils/ComboboxBase.types.ts:37]: I know, I just can't think of anything else 😅

**Pull request merged**